app/controllers/admin/page_elements_controller.rb

```diff
@@ -0,0 +1,59 @@
+# encoding:UTF-8
+class Admin::PageElementsController < ApplicationController
+  before_action :load_permissions
+  load_and_authorize_resource :page, parent: true, find_by: :url
+  load_and_authorize_resource :page_element, through: :page
+
+  def index
+    page = Page.includes(:page_elements).find_by!(url: params[:page_id])
+    @element_grid = initialize_grid(page.page_elements)
+  end
+
+  def new
+    page = Page.includes(:page_elements).find_by!(url: params[:page_id])
+    @page_element = page.page_elements.build
+  end
+
+  def edit
+    page = Page.includes(:page_elements).find_by!(url: params[:page_id])
+    @page_element = page.page_elements.find(params[:id])
+  end
```

Comment:
Could new and edit be the same action? The do the exact same thing.

Also, won't this trash any existing page_elements on the page?